lib/validation/validator.rb

```diff
@@ -1,132 +1,172 @@
 module Validation
-  module Rules
-    # A hash of rules for this object
-    def rules
-      @rules ||= {}
-    end
+  module Validator
+    attr_reader :object, :results
 
-    # A hash of errors for this object
-    def errors
-      @errors ||= {}
+    def initialize(object)
+      raise ArgumentError, "cannot validate nil" if object.nil?
```

Comment:
Agreed, we shouldn't arbitrarily restrict the functionality, even if 99.9% of the time it makes no sense to validate the methods of `nil`. I'll add syntax for validating the object itself soon.